include/vcpkg/sourceparagraph.h

```diff
@@ -38,16 +39,34 @@ namespace vcpkg
 
     struct Dependency
     {
+        struct Feature
```

Comment:
Robert had a comment that we generally should not use nested structs since we are accessing the nested struct outside. Also maybe we should rename to `DependencyFeature`?

Reply:
Sounds very generic. Maybe `FeatureOnPlatform` or `PlatformScopedFeature`or `FeatureWithPlatformExpr` :D